### PR TITLE
Updated some ValueError exceptions to throw InconsistentTableError.

### LINKED ISSUE #7413: Improve consistency of io.ascii exception classes
As a follow-on to #5578 there are a lot of exceptions in io.ascii that are currently `ValueError` which should really be `InconsistentTableError`.  Since the latter derives from `ValueError` this change could be made without any breakage.

### PR DESCRIPTION
**Pull request by humnaawan**
Fixes #7413

### CODE REVIEW

**Comment by bsipocz** [comment]
@taldcroft - We don't need a changelog for this, right?

**Comment by taldcroft** [comment]
@bsipocz - this seems like it deserves mention as an API change, although it should be back-compatible since `InconsistentTableError` derives from `ValueError`.

I haven't actually looked at this, but want to make sure it doesn't get in the way of #5578, which I think can be ready to merge pretty soon.

File: astropy/io/ascii/core.py [hunk 1]
```diff
@@ -30,6 +30,8 @@
 from ...utils.data import get_readable_fileobj
 from . import connect
 
+from . import core
```
Comment (taldcroft) [inline astropy/io/ascii/core.py]: Don't import yourself.

Comment (humnaawan) [reply astropy/io/ascii/core.py]: Oh, can you say how I should be calling `InconsistentTableError`?

Comment (humnaawan) [reply astropy/io/ascii/core.py]: I've removed the import; sorry, I didnt realize I was importing `core` in `core.py`.

**Comment by bsipocz** [comment]
@taldcroft - This is ready to go and would be nice to merge (as it was done during hackday PR-ing), but it's understandable if you want to held it up due to #5578

**Comment by taldcroft** [comment]
@bsipocz - can you check if this PR will introduce a conflict?  I'm just especially wary since #5578 is not actually rebase-able (at least by me) at this point.

File: CHANGES.rst [hunk 2]
```diff
@@ -134,6 +134,8 @@ astropy.extern
 astropy.io.ascii
 ^^^^^^^^^^^^^^^^
 
+- Updated some ValueError exceptions to throw InconsistentTableError. [#7425]
```
Comment (taldcroft) [inline CHANGES.rst]: You can give a bit more detail and guidance to users like so:
```
Changed a number of ``ValueError`` exceptions to ``InconsistentTableError`` in places
where the exception is related to parsing a table which is inconsistent with the
specified table format.  Note that ``InconsistentTableError`` inherits from ``ValueError``
so no user code changes are required. [#7425]
```

**Comment by bsipocz** [comment]
@taldcroft - If I try to cherry-pick these commits on top of the current #5578 then there is one tiny and trivial conflict that originates from a whitespace issue, but otherwise it's a clean on. 
I can give the rebase a try, it seemed very non-trivial at first, but maybe with some squashing first it can be simplified a bit.

**Comment by taldcroft** [comment]
@bsipocz - I did try the "squashing first" technique (which I generally find helpful), but strangely got the same conflicts.  I.e. I tried `git rebase -i <commit_before_first_5578_commit>`, and *still* got conflicts.  This made no sense.  Then I even tried squashing with `git rebase -i <first_5578_commit>`, and that gave (after a long wait) every commit from the master merge and I just aborted.

But your git fu may yet be powerful enough!!

**Comment by bsipocz** [comment]
Now that #5578 is merged, this PR indeed ended up with conflicts. I'm going ahead and rebase and address the comment above.

**Comment by bsipocz** [comment]
Thank you @humnaawan!

**Pull request merged**